modular_skyrat/modules/importedvendors/code/vendor_food.dm

```diff
@@ -0,0 +1,276 @@
+	icon_state = "foodtray_shroomfry"
+	tastes = list("mushroom" = 4, "becoming rich" = 1, "laziness" = 1)
+	foodtypes = VEGETABLES
+
+/obj/effect/spawner/random/vendor_meal_sides
+	name = "random side spawner"
+	desc = "I hope I get one that actually matches my meal."
+	icon_state = "loot"
+
+/obj/effect/spawner/random/vendor_meal_sides/nt
+	name = "random nt side spawner"
+
+/obj/effect/spawner/random/vendor_meal_sides/nt/Initialize(mapload)
+	loot = list(
+		/obj/item/food/vendor_tray_meal/side,
+		/obj/item/food/vendor_tray_meal/side/crackers_and_jam,
+		/obj/item/food/vendor_tray_meal/side/crackers_and_cheese,
+	)
+	. = ..()
+
+/obj/item/food/vendor_tray_meal/side
+	name = "\improper NT-Side: Flatbread and Peanut Butter"
+	desc = "A small stack of tough flatbread, and a small spread of peanut butter for each."
+	icon_state = "foodpack_nt"
+	trash_type = /obj/item/trash/empty_side_pack/nt
+	food_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	tastes = list("tough bread" = 2, "peanut butter" = 2)
+	foodtypes = GRAIN
+	hot_and_steamy = FALSE
+	custom_price = PAYCHECK_LOWER * 2.5
+
+/obj/item/food/vendor_tray_meal/side/crackers_and_jam
+	name = "\improper NT-Side: Flatbread and Berry Jelly"
+	desc = "A small stack of tough flatbread, and a small spread of nondescript berry jelly for each."
+	tastes = list("tough bread" = 2, "berries" = 2)
+	foodtypes = GRAIN | FRUIT
+
+/obj/item/food/vendor_tray_meal/side/crackers_and_cheese
+	name = "\improper NT-Side: Flatbread and Cheese Spread"
+	desc = "A small stack of tough flatbread, and a small spread of cheese for each."
+	tastes = list("tough bread" = 2, "cheese" = 2)
+	foodtypes = GRAIN | DAIRY
+
+/obj/effect/spawner/random/vendor_meal_sides/yangyu
+	name = "random yangyu side spawner"
+
+/obj/effect/spawner/random/vendor_meal_sides/yangyu/Initialize(mapload)
+	loot = list(
+		/obj/item/food/vendor_tray_meal/side/miso,
+		/obj/item/food/vendor_tray_meal/side/rice,
+		/obj/item/food/vendor_tray_meal/side/pickled_vegetables,
+	)
+	. = ..()
+
+/obj/item/food/vendor_tray_meal/side/miso
+	name = "\improper 副菜: Miso Soup"
+	desc = "This is quite literally just a plastic bag full of miso soup, remember to only open it on the side facing up."
+	icon_state = "foodpack_yangyu"
+	trash_type = /obj/item/trash/empty_side_pack/yangyu
+	tastes = list("miso" = 2)
+	foodtypes = VEGETABLES
+
+/obj/item/food/vendor_tray_meal/side/rice
+	name = "\improper 副菜: White Rice"
+	desc = "A bag stuffed full of white rice, in case your meal didn't come with enough to sate your needs."
+	icon_state = "foodpack_yangyu"
+	trash_type = /obj/item/trash/empty_side_pack/yangyu
+	tastes = list("old rice" = 2)
+	foodtypes = GRAIN
+
+/obj/item/food/vendor_tray_meal/side/pickled_vegetables
+	name = "\improper 副菜: Pickled Vegetables"
+	desc = "Contains a small assortment of vegetables picked in a vinegar-like solution."
```

Comment:
Just to check, is "picked" the right word or is it meant to be "packed" or "pickled"?

Reply:
it is meant to be pickled, I coded this bit at like 2 am :(